src/utilities/linkHighlightedJson.js

```diff
@@ -0,0 +1,83 @@
+import _ from 'lodash';
+import {xdrViewer, singleAccount} from './linkBuilder';
+
+// Map of functions specific to JSON property keys that creates urls for a given
+// value.
+// A function can also return undefined if it doesn't deem the value worthy of
+// linking (for example: invalid account ID or keys that have ambiguous names)
+let accountIdGenerator = (value) => {
+  if (value.length === 56) {
+    return singleAccount(value);
+  }
+};
+let valueUrlGenerator = {
+  href: (value) => {
+    return value;
+  },
+  'envelope_xdr': (value) => {
+    return xdrViewer(value, 'TransactionEnvelope');
+  },
+  'result_xdr': (value) => {
+    return xdrViewer(value, 'TransactionResult');
+  },
+  'result_meta_xdr': (value) => {
+    return xdrViewer(value, 'TransactionMeta');
+  },
+  'id': accountIdGenerator,
+  'public_key': accountIdGenerator,
+  'account_id': accountIdGenerator,
+  'funder': accountIdGenerator,
+  'account': accountIdGenerator,
+  'source_account': accountIdGenerator,
+  'destination_account': accountIdGenerator,
+};
+
+// linkHighlightedJson annotates and adds relevant links to json code highlighted
+// by prism.js.
+
+// @param {string} code - Valid html generated by prism.js in JSON language mode
+export default function linkHighlightedJson(code) {
+  let parser = new DOMParser();
+  let document = parser.parseFromString(code, 'text/html');
+  let body = document.childNodes[0].childNodes[1];
+
+  // Edit the token elements in place
+  _.each(body.children, (token, index) => {
+    if (token.className.indexOf('property') > 0) {
+      annotatePropertyToken(token);
+    }
+  });
+  return body.innerHTML;
+}
+
+// Takes in a property highlighted item. If functionality is defined for it's
+// corresponding property (only string type supported) and the content is valid,
+// then we will turn the value node into a link.
+// @param {HTMLElement} propertyToken - The token element of property type (assigned by prism)
+function annotatePropertyToken(propertyToken) {
+  let valueToken = propertyToken.nextElementSibling.nextElementSibling;
```

Comment:
As far as I can tell, there isn't a way to iterate over react elements (I'm turning a html string into a group of react elements) so I have to do my chances to the highlighted code one of three ways (there may be another that I do not know of though).

1. Edit the HTML string using DOMParser before giving the HTML string to React
2. Edit the HTML string using DOMParser after giving the HTML string to React by using `componentDidUpdate`. This would be more messy since now the React element output is different from what is the final output.
3. Write a PrismJS plugin to add links to JSON code. An example is in the prism-autolinker plugin which does a related thing: https://github.com/PrismJS/prism/blob/gh-pages/plugins/autolinker/prism-autolinker.js

I chose to do option 1 (the current code) instead of writing a PrismJS plugin since I thought it would be simpler. My current implementation takes 45 lines of code that I believe to not be dense  and not too complicated as opposed to the Prism API and it's data structures.

It was the expedient choice over the "right" solution which is to build a prismjs plugin. I would love to write the plugin and even contribute it to `prism` since I think it's a feature that could help many. But for now, I chose the "easy way" by keeping it as a separate layer.

I think the highlighting feature is here to stay, but i'm not sure I could find a better solution considering the tradeoffs.